tools/fpgadiag/fpgadiag

```diff
@@ -40,19 +40,22 @@ cmd_map = {'lpbk1': ['nlb0'],
            'mb1': ['mb1']}
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
+    parser = argparse.ArgumentParser(add_help=False)
     parser.add_argument('-t', '--target',
                         default='fpga',
                         choices=['fpga', 'ase'],
                         help='choose target')
     parser.add_argument('-m', '--mode',
                         choices=cmd_list,
-                        required=True,
                         help='choose test mode')
 
     args, leftover = parser.parse_known_args()
 
-    cmdline = cmd_map[args.mode] + leftover
+    if args.mode:
+        cmdline = cmd_map[args.mode] + leftover
```

Comment:
Does this print help if "-m" is set?

Reply:
Yes. If "-m" is given, "-h" or "--help" print detailed help info for the test. If "-m" is not given, "-h" or "--help" print a high-level message about the command line syntax, and describing what modes (tests) and targets are available.

Reply:
Github cuts off my sentence. But the complete sentence can be seen in the commit message.

Reply:
When I run 
<fpgadiag -m -h>
should I expect a  more in depth description of lbk1, read, write, trput, sw, and mb1?
Right now I get the same thing for the aforementioned command line and
<fpgadiag -t -h>

usage: fpgadiag [-t {fpga,ase}] [-m {lpbk1,read,write,trput,sw,mb1}]

Reply:
You will get an in depth description for a mode if the mode is specified. So for example,
`fpgadiag -m lpbk1 -h` will give a description for mode `lpbk1`.

Reply:
so if i  understand correctly the expected behavior for running is as follows
```
[jabuie@sj-avl-d22-mc ~]$ fpgadiag -m write -h
Usage: nlb3 [options]

    --help, -h. Show help
    --config, -c. Path to test config file Default=nlb3.json
    --target, -t. one of { fpga, ase } Default=fpga
    --mode, -m. mode { read, write, trput } Default=write
    --begin, -b. where 1 <= <value> <= 65535 Default=1
    --end, -e. where 1 <= <value> <= 65535 Default=1
    --multi-cl, -u. one of {1, 2, 4} Default=1
    --strided-access, -a. where 1 <= <value> <= 64 Default=1
```
the only thing i would suggest would be to have a message a message showing the options for the help flag. Right now looking at just fpgadiag the only optional args are `-t` and `-m`. I think users would appreciate some indication of how helpful the -h flag can be.

Reply:
Thanks for the comment. I just pushed another review that incorporates the feedback.